### PR TITLE
Implement --json flag for changing report types

### PR DESCRIPTION
**Pull request by Michionlion**
This implements the `--json` flag that will switch the report type to JSON. The old implementation of `output_json` (already written by @gkapfham) provides both the index of the check, and the check results themselves. The new implementation simply adds each of the check result 'objects' to a list which is then printed, resulting in the following format (whitespace is not guarenteed):

```
{
  "check": "Description of check",
  "outcome": true,
  "diagnostic": ""
}

{
  "check": "Another check",
  "outcome": false,
  "diagnostic": "Description of failure"
}
```

### CODE REVIEW

**Comment by Michionlion** [comment]
@gkapfham, is there a reason for the original implementation of `output_json` to add only one entry to the returned list if there were multiple results?

**Comment by gkapfham** [comment]
Hi @Michionlion, my recollection of the `output_json` is as follows: if you give it a report with a single "row" in it, it will render that as JSON. This is what I expect would normally be used since the expected convention is that GatorGrader will be called to perform a *single* check each time in the GatorGradle configuration. Also, I think that even if someone were to use GatorGrader interactively, it would be expected that each call on the command-line would perform a *single* check.

As such, I think that there is not a need to have the report be a dictionary of dictionaries, which is what, if I recall correctly, it was previously. This makes me think that the implementation I created was "over-engineered" in that it was designed to also handle multiple checks in a single run. I have a nagging doubt that there is some benefit to that nested that I am overlooking right now. But, I think that it is safe to go with this non-nested approach that you are considering.

Okay, does this make sense? Let me know if you have further clarifying questions.

**Comment by gkapfham** [comment]
@Michionlion, I also have one clarifying comment about your first comment on this PR. You have written `"outcome": true` and I think that this would actually be `"outcome": True` since the value stored in the dictionary is a Python boolean and this is represented as `True`. I know that this is really small, but I wanted to confirm the matter.

**Comment by gkapfham** [comment]
@Michionlion, one final small suggestion for a check: I think that we should make sure that the symbols (i.e., the check-mark and the x-mark) are both correctly appearing in either the message or the diagnostic field. I'm pretty sure that I am storing this in the report, but wanted to but this note here as a reminder to ensure that we check this before merging the PR.

**Comment by Michionlion** [comment]
I think the true/false values are actually written as `true` and `false` when serializing to JSON, but the way that it is handled in parsing should aleivate any problems -- any combination of capitalization is supposedly valid JSON. For me at least, the check, cross, and arrow are correctly displaying on all systems I have access to.

**Comment by gkapfham** [comment]
@Michionlion, thanks for the confirmation about these issues. I am going to merge this PR into master. Then, I plan to add some test cases to (slightly) improve coverage and rephrase one of the comments. Once all of those small changes are finished, I will start to improve the documentation and do integration testing between GatorGrader and GatorGradle.

File: gator/orchestrate.py [hunk 1]
```diff
@@ -36,6 +39,10 @@ def check_arguments(system_arguments):
     # Action: display the welcome message
     if gg_arguments.nowelcome is not True:
         actions.append([DISPLAY, "welcome_message", VOID])
+    if gg_arguments.json is True:
+        # pylint: disable=global-statement
+        global OUTPUT_TYPE
+        OUTPUT_TYPE = getattr(REPORT, "JSON")
```
[the remark this answers is not in the document]
Comment (Michionlion) [reply gator/orchestrate.py]: Thanks for taking care of this issue! I was unsure how to test this with the pytest framework you had any set up, so I held off on the hopes that you could point me in the right direction. :)

**Pull request merged**